Examples/Tests/PythonWrappers/PICMI_inputs_2d.py

```diff
@@ -0,0 +1,262 @@
+    Bypml = pwxf.ByFPPMLWrapper(include_ghosts = include_ghosts)
+    Bzpml = pwxf.BzFPPMLWrapper(include_ghosts = include_ghosts)
+    Fpml  = pwxf.FFPPMLWrapper(include_ghosts = include_ghosts)
+    Gpml  = pwxf.GFPPMLWrapper(include_ghosts = include_ghosts)
+
+# Compute min and max of fields data
+def compute_minmax(data):
+    vmin = -max(abs(data.min()), abs(data.max()))
+    vmax =  max(abs(data.min()), abs(data.max()))
+    return (vmin, vmax)
+
+# Plot fields data either in valid domain or in PML
+def plot_data(it, data, edge, pml, title, name):
+    fig, ax = plt.subplots(nrows = 1, ncols = 1, gridspec_kw = dict(wspace = 0.5), figsize = [6,5])
+    cax = make_axes_locatable(ax).append_axes('right', size='5%', pad='5%')
+    vmin, vmax = compute_minmax(data)
+    im = ax.imshow(data, origin = 'lower', extent = edge, vmin = vmin, vmax = vmax, cmap = 'seismic')
+    ax.set_xlabel('x')
+    ax.set_ylabel('z')
+    ax.set_title(title)
+    fig.colorbar(im, cax = cax)
+    # Draw PMLs and ghost regions
+    lw  = 0.8
+    ls = '-'
+    if (pml):
+        ax.axvline(x = 0       , linewidth = lw, color = 'grey' , linestyle = ls)
+        ax.axvline(x = 0+nxg   , linewidth = lw, color = 'black', linestyle = ls)
+        ax.axvline(x = -nxpml  , linewidth = lw, color = 'grey' , linestyle = ls)
+        ax.axvline(x = nx      , linewidth = lw, color = 'grey' , linestyle = ls)
+        ax.axvline(x = nx-nxg  , linewidth = lw, color = 'black', linestyle = ls)
+        ax.axvline(x = nx+nxpml, linewidth = lw, color = 'grey' , linestyle = ls)
+        ax.axhline(y = 0       , linewidth = lw, color = 'grey' , linestyle = ls)
+        ax.axhline(y = 0+nzg   , linewidth = lw, color = 'black', linestyle = ls)
+        ax.axhline(y = -nzpml  , linewidth = lw, color = 'grey' , linestyle = ls)
+        ax.axhline(y = nz      , linewidth = lw, color = 'grey' , linestyle = ls)
+        ax.axhline(y = nz-nzg  , linewidth = lw, color = 'black', linestyle = ls)
+        ax.axhline(y = nz+nzpml, linewidth = lw, color = 'grey' , linestyle = ls)
+    else:
+        ax.axvline(x = 0       , linewidth = lw, color = 'grey' , linestyle = ls)
+        ax.axvline(x = nx      , linewidth = lw, color = 'grey' , linestyle = ls)
+        ax.axhline(y = 0       , linewidth = lw, color = 'grey' , linestyle = ls)
+        ax.axhline(y = nz      , linewidth = lw, color = 'grey' , linestyle = ls)
+    # Plot title
+    suptitle = 'PML in (x,z), 4 grids 64 x 64'
+    plt.suptitle(suptitle)
+    figname = 'figure_' + name + '.png'
+    fig.savefig(figname, dpi = 100)
+
+# Initialize fields data (unit pulse) and apply smoothing
+def init_data(data):
+    data[nx//2,nz//2] = 1.
+    data[2:nx-1,:] *= 0.5
+    data[2:nx-1,:] += 0.5 * (data[1:nx-2,:] + data[3:nx,:])
+    data[:,2:nz-1] *= 0.5
+    data[:,2:nz-1] += 0.5 * (data[:,1:nz-2] + data[:,3:nz])
+
+# Advance simulation for one time step and get fields data
+sim.step(1)
```

Comment:
Instead of doing one step, you could also do

> sim.initialize_inputs()
sim.initialize_warpx()

Though that has the hazard that the `step` routine may change. (Though it would be best if the `Simulation` class had a separate initialization routine, but that would be a different PR.)

Reply:
I prefer the way you suggested, so I fixed this in https://github.com/ECP-WarpX/WarpX/pull/2576/commits/c5c06f4115254f07762595c32b2d6b83a1fc2379.